features/step_definitions/barclamp_keystone/keystone_steps.rb

```diff
@@ -0,0 +1,54 @@
+Given(/^the chef role "([^"]*)" exists on admin node$/) do |arg1|
+    keystone_server_role = admin_node.exec!("knife role show keystone-server").output
+    expect(keystone_server_role).not_to be_empty
+end
+
+And(/^the "([^"]*)" cookbook exists on the admin node$/) do |arg1|
+    keystone_cookbook = admin_node.exec!("knife cookbook show keystone").output
+    expect(keystone_cookbook).not_to be_empty
+end
+
+And(/^the barclamp proposal is using the keystone identity service$/) do
+    json_response = JSON.parse(admin_node.exec!("crowbar keystone show default").output)
+    @node_fqdn = json_response["deployment"]["keystone"]["elements"]["keystone-server"][0]
+    expect(@node_fqdn).not_to be_empty
+end
+
+When(/^the node with keystone-server role has been detected successfully$/) do
+    user_list = control_node.openstack.user.list
+    expect(user_list).not_to be_empty
+end
+
+And(/^the "([^"]*)" is installed in the controller node$/) do |test_package|
+    control_node.rpm_q(test_package)
+end
+
+And(/^the "([^"]*)" is installed on the controller node$/) do |client_package|
+    control_node.rpm_q(client_package)
+end
+
+Then(/^I can create a new project "([^"]*)" on the cloud$/) do |project_name|
```

Comment:
The client uses SSL and PKI tokens if that's what was deployed. AFAIR we agreed to not check things like "Feature A was enabled, now check if Feature A is really enabled". That will be done with ```crowbar_batch```. Or am I missunderstanding your question?